### PR TITLE
Data-Driven Keyboard Conversions: I

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

Converts `rules.mk` entries to data-driven where applicable. Renames `info.json` to `keyboard.json` in instances where `rules.mk` can be safely deleted.

- keyboards with names beginning with `i`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ibm/model_m_4th_gen/overnumpad_1xb/rules.mk [hunk 1]
```diff
@@ -1,17 +1,3 @@
...
-EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = no         # Console for debug
-COMMAND_ENABLE = no         # Commands for debug and configuration
-NKRO_ENABLE = no            # Enable N-Key Rollover
-BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
-ENCODER_ENABLE = no         # Enable rotary encoder support
-AUDIO_ENABLE = no           # Audio output
-KEYBOARD_SHARED_EP = yes    # Free up some extra endpoints - needed if console+mouse+extra
+KEYBOARD_SHARED_EP = yes
```
Comment (fauxpark) [inline keyboards/ibm/model_m_4th_gen/overnumpad_1xb/rules.mk:1]: Can be migrated.

File: keyboards/inland/kb83/rules.mk [hunk 2]
```diff
@@ -1,17 +1,2 @@
-# Build Options
-#   change yes to no to disable
-#
-BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
-MOUSEKEY_ENABLE = no        # Mouse keys
-EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = no         # Console for debug
-COMMAND_ENABLE = no         # Commands for debug and configuration
 KEYBOARD_SHARED_EP = no
```
Comment (fauxpark) [inline keyboards/inland/kb83/rules.mk:1]: Can be migrated.

**Review by fauxpark — APPROVED** [review]

File: keyboards/inland/kb83/rules.mk (hunk 2 above)
Comment (noroadsleft) [reply keyboards/inland/kb83/rules.mk:1]: Should I migrate this as `usb.shared_endpoint.keyboard=false`, or simply delete this line?

Comment (fauxpark) [reply keyboards/inland/kb83/rules.mk:1]: Didn't see it was `no`, just delete it.

**Review by zvecr — APPROVED** [review]

**Pull request merged**